src/types.jl

```diff
@@ -1,32 +1,56 @@
-struct SDEKernel{fType,gType,tType,dtType,paramType1,paramType2}
+struct SDEKernel{fType,gType,tType,pType}
   f::fType
   g::gType
-  tstart::tType
-  tend::tType
-  dt::dtType
-  pest::paramType1
-  plin::paramType2
+  trange::tType
+  p::pType
 end
 
-function SDEKernel(f,g,tstart,tend,pest,plin;dt=nothing)
-  SDEKernel{typeof(f),typeof(g),typeof(tstart),
-            typeof(dt),typeof(pest),typeof(plin)}(f,g,tstart,tend,dt,pest,plin)
+struct Message{kernelType,solType,sol2Type,tType}
+  ktilde::kernelType
+  sol::solType
+  soldis::sol2Type
+  ts::tType
 end
```

Comment:
Nice. Quoting our paper:  while other choices for the message are sometimes convenient, they are essentially equivalent: the format of the message `m` is something private between the forward map and the backward map. Having a type here makes sure that both sides agree on the interpretation